notebooks/DiCE_getting_started_feasible.ipynb

```diff
@@ -0,0 +1,1072 @@
+    "        optimizer.zero_grad()\n",
+    "        train_x= torch.tensor( train_batches[i][:,:-1] ).float() \n",
+    "        train_y= torch.tensor( train_batches[i][:,-1], dtype=torch.int64 )\n",
+    "        \n",
+    "        out= m(train_x)\n",
+    "        train_acc += torch.sum( torch.argmax(out, axis=1) == train_y )\n",
+    "        \n",
+    "        # Cross Entropy Loss\n",
+    "        loss= crieterion(out, train_y)\n",
+    "        #L2 Regularization\n",
+    "        weight_norm = torch.tensor(0.)\n",
+    "        for w in m.ann_model.parameters():\n",
+    "            weight_norm += w.norm().pow(1)\n",
+    "        loss+= 0.001*weight_norm\n",
+    "        \n",
+    "        loss.backward()\n",
+    "        optimizer.step()\n",
+    "    print(train_acc, len(train_dataset))     "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Validation        \n",
+    "np.random.shuffle(val_dataset)\n",
+    "train_batches= np.array_split( val_dataset, val_dataset.shape[0]//batch_size ,axis=0 )    \n",
+    "val_acc=0.0\n",
+    "for i in range(len(train_batches)):    \n",
+    "    optimizer.zero_grad()\n",
+    "    train_x= torch.tensor( train_batches[i][:,:-1] ).float() \n",
+    "    train_y= torch.tensor( train_batches[i][:,-1], dtype=torch.int64 )\n",
+    "    out= m(train_x)\n",
+    "    val_acc += torch.sum( torch.argmax(out, axis=1) == train_y )\n",
+    "print(val_acc, len(val_dataset))\t\n",
+    "\n",
+    "#Saving the Black Box Model\n",
+    "base_model_dir= '../dice_ml/utils/sample_trained_models/'\n",
+    "dataset_name= 'adult'\n",
+    "path=base_model_dir+dataset_name+'-pytorch.pth'\n",
+    "torch.save(m.state_dict(), path)   "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Generate feasible counterfactuals"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Based on the data object *d* and the model object *m*, we can now instantiate the DiCE class for generating explanations. \n",
+    "We present the variational inference based approach towards generating counterfactuals, where we first train an encoder-decoder framework to generate counterfactuals. More details about our framework can be found here:https://arxiv.org/abs/1912.03277\n",
+    "\n",
+    "DiceBaseGenCF class has an attribute .train(), which would train the Variational Encoder Decoder framework on the input dataframe d. It has another arugment, 'pre_trained', which if set to 0 would train the framework for generating CF. Else, it can be set to 1 to avoid repreated training of the framework and would load the latest optimal model "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "from dice_ml.dice_interfaces.dice_base_gencf import DiceBaseGenCF"
```

Comment:
Can you import DiceBaseGenCF as how it is suggested here https://github.com/interpretml/DiCE/blob/master/dice_ml/model.py#L13. In your case, you might need to give to backend this dict `{'model':'pytorch_model.PyTorchModel', 'explainer':'dice_base_gencf.DiceBaseGenCF'}`
Tha way, you can simply do `exp = dice_ml.Dice(d, m)' to automatically import DiceBaseGenCF

Reply:
Currently, we need to set the backend via the model.py class that leads to calling the method multiple times for different explainers (BaseGenCF, ModelApproxGenCF). Need to update it.